wezterm-gui/src/overlay/copy.rs

```diff
@@ -125,6 +127,17 @@ impl CopyOverlay {
             .clone()
             .ok_or_else(|| anyhow::anyhow!("failed to clone window handle"))?;
         let dims = pane.get_dimensions();
+        let pattern = if params.pattern.is_empty() {
+            SAVED_PATTERN
+                .lock()
+                .get(&tab_id)
+                .map(|p| p.clone())
+                .unwrap_or(params.pattern)
+        } else {
+            params.pattern
+        };
+        let search_line = LineEditBuffer::new(&pattern, unicode_column_width(&pattern, None));
```

Comment:
```suggestion
        let search_line = LineEditBuffer::new(&pattern, pattern.len());
```
It seems this can resolve the issue of the search overlay crashing upon reopening. https://github.com/wez/wezterm/pull/5416#issuecomment-2164280168

Reply:
```rust
/// Returns the number of cells visually occupied by a sequence
/// of graphemes.
/// Calls through to `grapheme_column_width` for each grapheme
/// and sums up the length.
pub fn unicode_column_width(s: &str, version: Option<UnicodeVersion>) -> usize {
```
and
```rust
pub struct LineEditBuffer {
    line: String,
    /// byte index into the UTF-8 string data of the insertion
    /// point.  This is NOT the number of graphemes!
    cursor: usize,
}
```

@Mrreadiness 
I think I have found the cause of the incorrect cursor position.

The cursor field expects a byte index.
But `unicode_column_width` returns the grapheme width(visual width) of characters. 
For example, a Chinese character has a visual width of 2 English characters but is actually stored as 3 bytes.

Reply:
@kenchou Thanks! You are right. On this line using `unicode_column_width` is incorrect.